example/push/Main.hs

```diff
@@ -0,0 +1,52 @@
+module Main where
+
+import Control.Exception (displayException)
+import Data.Aeson ((.=))
+import Data.Monoid ((<>))
+
+import qualified Data.Aeson as A
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.HashMap.Strict as H
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import qualified Data.Yaml as Y
+import qualified Network.Pusher as P
+import qualified Network.Pusher.Protocol as P
+
+main :: IO ()
+main = do
+  eitherCred <- Y.decodeFileEither "../credentials.yaml"
+  case eitherCred of
+    Left e -> print e
+    Right cred -> do
+      pusher <- P.getPusher cred
+      demoPushNotification pusher
+
+demoPushNotification :: P.Pusher -> IO ()
+demoPushNotification pusher = do
+  let Just interest = P.mkInterest "exampleinterest"
+      fcmObject :: A.Object
+      fcmObject =
+        H.fromList
+          [ ( "notification"
+            , A.Object $
+              H.fromList
+                [ ("title", A.String "Sup?")
+                , ("body", A.String "Hello Firebase.")
+                , ("icon", A.String "firebase-logo.png")
+                ])
+          ]
+      notification =
+        P.Notification
+        { P.notificationInterest = interest
+        , P.notificationWebhookURL = Nothing
+        , P.notificationWebhookLevel = Nothing
+        , P.notificationAPNSPayload = Nothing
+        , P.notificationGCMPayload = Nothing
+        , P.notificationFCMPayload = Just $ P.FCMPayload fcmObject
+        }
```

Comment:
This is quite clunky for the end user. I'm assuming in most cases only a couple of these would be set. It might be worth thinking about helper functions that could set less frequently used (e.g. webook fields) to Nothing, or only set a single payload. Unfortunately the builder pattern doesn't seem to be straightforward in Haskell, it might be worth looking into though.

This shouldn't stop us merging this for now though :)